wp-content/themes/core/components/blocks/stats/Stats_Block_Controller.php

```diff
@@ -0,0 +1,162 @@
+		$this->content_classes   = (array) $args[ self::CONTENT_CLASSES ];
+		$this->classes           = (array) $args[ self::CLASSES ];
+		$this->attrs             = (array) $args[ self::ATTRS ];
+		$this->stats             = (array) $args[ self::STATS ];
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function defaults(): array {
+		return [
+			self::LAYOUT            => Stats_Block::LAYOUT_STACKED,
+			self::CONTENT_ALIGN     => Stats_Block::CONTENT_ALIGN_LEFT,
+			self::DISPLAY_DIVIDERS  => Stats_Block::DISPLAY_DIVIDERS,
+			self::TITLE             => '',
+			self::DESCRIPTION       => '',
+			self::CONTAINER_CLASSES => [ 'b-stats__container', 'l-container' ],
+			self::CONTENT_CLASSES   => [ 'b-stats__content' ],
+			self::CLASSES           => [ 'c-block', 'b-stats' ],
+			self::ATTRS             => [],
+			self::STATS             => [],
+		];
+	}
+
+	/**
+	 * @param array $args
+	 *
+	 * @return array
+	 */
+	public function get_content_args(): array {
+		return [
+			'tag'     => 'header',
+			'classes' => [ 'b-stats__header' ],
+			'title'   => defer_template_part( 'components/text/text', null, [
+				'content' => $this->title,
+				'tag'     => 'h2',
+				'classes' => [ 'b-stats__title', 'h3' ],
+			] ),
+			'content' => defer_template_part( 'components/text/text', null, [
+				'content' => $this->description,
+				'classes' => [ 'b-stats__description', 't-sink', 's-sink' ],
+			] ),
+			'layout'  => Stats_Block::CONTENT_ALIGN_CENTER === $this->content_align ? Content_Block::LAYOUT_CENTER : Content_Block::LAYOUT_STACKED,
+		];
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_stats(): array {
+		$statistic_args = [];
+
+		foreach ( $this->stats as $item ) {
+			// Skip over statistic rows with no value.
+			if ( empty( $item[ Stats_Block::ROW_VALUE ] ) ) {
+				continue;
+			}
+
+			$statistic_args[] = [
+				Statistic::VALUE => defer_template_part( 'components/text/text', null, $this->get_value_args( $item ) ),
+				Statistic::LABEL => defer_template_part( 'components/text/text', null, $this->get_label_args( $item ) ),
+			];
+		}
+
+		return $statistic_args;
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function get_value_args( $item ): array {
+		return [
+			'content' => esc_html( $item[ Stats_Block::ROW_VALUE ] ),
+		];
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function get_label_args( $item ): array {
+		return [
+			'content' => esc_html( $item[ Stats_Block::ROW_LABEL ] ),
+		];
+	}
+
+	/**
+	 * @return string
+	 */
+	public function container_classes(): string {
+		return Markup_Utils::class_attribute( $this->container_classes );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function content_classes(): string {
+		return Markup_Utils::class_attribute( $this->content_classes );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function classes(): string {
+		$this->classes[] = 'c-block--' . $this->layout;
+		$this->classes[] = 'c-block--' . $this->display_dividers;
```

Comment:
Unsure of the above class, but I'd go `b-stats--` vs. `c-block--` as the base prefix for this particular class as it is very contextually based on the stats block vs. a globally needed block thing